bin/convert-formula.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env sh
+set -o nounset # Treat unset variables as an error and immediately exit
+set -o errexit # If a command fails exit the whole script
+
+if [ "${DEBUG:-false}" = "true" ]; then
+  set -x # Run the entire script in debug mode
+fi
+
+usage() {
+  echo "usage: $(basename "$0") <new-formula-name>" 1>&2
+  echo 1>&2
+  echo "Convert template-formula to <new-formula-name>-formula." 1>&2
+  echo "<new-formula-name> should be a string of lowercase characters and numbers only." 1>&2
+  echo "<new-formula-name> should not be any of 'bin' 'docs' 'test'." 1>&2
+}
+
+args() {
+  if [ $# -ne 1 ]; then
+    usage
+    exit 1
+  fi
+  NEW_NAME=$1
+  if echo "$NEW_NAME" | grep -E --quiet --invert-match '^[a-z0-9]+$'; then
+    usage
+    exit 1
+  fi
+  if echo bin docs test | grep --quiet --word-regexp "$NEW_NAME"; then
+    usage
+    exit 1
+  fi
+}
+
+convert_formula() {
+  git reset \
+    "$(echo "feat: initial commit" \
+      | git commit-tree "HEAD^{tree}")"
+  git rm bin/convert-formula.sh AUTHORS.md CHANGELOG.md
+  git mv template__  "${NEW_NAME}"
+  grep -rl template__ . --exclude-dir=.git | xargs perl -pi -e s/template__/"${NEW_NAME}"/g
```

Comment:
@dafyddj This may help with the `perl` vs. `sed` issue:

* https://github.com/saltstack/salt-bootstrap/commit/8647b74cbc7f134b22d8b5ab6f9975440736dc37

Reply:
Thanks, but the problematic flag is `-i` as explained [here](https://stackoverflow.com/q/5694228/5009408) but as I look closer there may be an another way further down in the answers on that page.

Reply:
Resolved with `ex`.